stl/src/vector_algorithms.cpp

```diff
@@ -4947,145 +5012,267 @@ namespace {
+                                _Advance_bytes(_First1, _Pos * sizeof(_Ty));
+                                // Full match because size is less than 16. Return this match.
+                                return _First1;
+                            }
                         }
+
+                        return _Last1;
+                    } else { // _Size_bytes_2 is greater than 16 bytes
+                        const __m128i _Data2  = _mm_loadu_si128(reinterpret_cast<const __m128i*>(_First2));
+                        const size_t _Max_pos = _Size_bytes_1 - _Size_bytes_2;
+
+                        const void* _Stop1 = _First1;
+                        _Advance_bytes(_Stop1, _Max_pos);
+
+                        const void* _Tail2 = _First2;
+                        _Advance_bytes(_Tail2, 16);
+
+                        do {
+                            const __m128i _Data1 = _mm_loadu_si128(static_cast<const __m128i*>(_First1));
+                            if (!_mm_cmpestrc(_Data2, _Part_size_el, _Data1, _Part_size_el, _Op)) {
+                                _Advance_bytes(_First1, 16); // No matches, next.
+                            } else {
+                                const int _Pos = _mm_cmpestri(_Data2, _Part_size_el, _Data1, _Part_size_el, _Op);
+
+                                bool _Match_1st_16 = true;
+
+                                if (_Pos != 0) {
+                                    _Advance_bytes(_First1, _Pos * sizeof(_Ty));
+
+                                    if (_First1 > _Stop1) {
+                                        break; // Oops, doesn't fit
+                                    }
+
+                                    // Match not from the first byte, check 16 symbols
+                                    const __m128i _Match1 = _mm_loadu_si128(static_cast<const __m128i*>(_First1));
+                                    const __m128i _Cmp    = _mm_xor_si128(_Data2, _Match1);
+                                    if (!_mm_testz_si128(_Cmp, _Cmp)) {
+                                        _Match_1st_16 = false;
+                                    }
+                                }
+
+                                if (_Match_1st_16) {
+                                    const void* _Tail1 = _First1;
+                                    _Advance_bytes(_Tail1, 16);
+
+                                    if (memcmp(_Tail1, _Tail2, _Size_bytes_2 - 16) == 0) {
+                                        return _First1;
+                                    }
+                                }
+
+                                // Start from the next element
+                                _Advance_bytes(_First1, sizeof(_Ty));
+                            }
+                        } while (_First1 <= _Stop1);
                     }
 
-                    if (_Equal) {
-                        return _Ptr1;
+                    return _Last1;

```

Comment:
Style: The position of this `return` is now slightly strange. Above, `if (_Size_bytes_2 <= 16) {` ends with `return _Last1;`, but the `else` falls through only to `return _Last1;` here.

This is inconsistent with the AVX2 codepath above, as well as both the AVX2 and SSE4.2 codepaths in `_Find_end_impl` below.